test/react/functional-components/dynamic-type.js

```diff
@@ -0,0 +1,28 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+
+function Foo(props) {
+  return <span>{props.name}</span>;
+}
+
+function Bar(props) {
+  return <div>{props.name}</div>;
+}
+
+function App(props) {
+  let Type = props.switch ? Foo : Bar;
+
+  return <Type name={"Dominic"} />
+}
+
+App.getTrials = function(renderer, Root) {
+  renderer.update(<Root />);
+  return [['render with dynamic prop access', renderer.toJSON()]];
```

Comment:
This test should verify updates work too. (Particularly switching the `switch`.)

Reply:
Will add more test cases.